diff --git a/lib/commands/pack/win.js b/lib/commands/pack/win.js
index 488e61a50e3183a2183b69f8a8571cb9523bdae4..f71e9d9d2c87a5b8aef4f672ac9cb01398f3e4dc 100644
--- a/lib/commands/pack/win.js
+++ b/lib/commands/pack/win.js
@@ -37,12 +37,13 @@ const scripts = {
     /* eslint-disable no-useless-escape */
     cmd: (config, additionalCLI) => `@echo off
 setlocal enableextensions
+set NODE_NO_WARNINGS=1
 
 set ${additionalCLI ? `${additionalCLI.toUpperCase()}_BINPATH` : config.scopedEnvVarKey('BINPATH')}=%~dp0\\${additionalCLI ?? config.bin}.cmd
 if exist "%LOCALAPPDATA%\\${config.dirname}\\client\\bin\\${additionalCLI ?? config.bin}.cmd" (
   "%LOCALAPPDATA%\\${config.dirname}\\client\\bin\\${additionalCLI ?? config.bin}.cmd" %*
 ) else (
-  "%~dp0\\..\\client\\bin\\node.exe" "%~dp0\\..\\client\\${additionalCLI ? `${additionalCLI}\\bin\\run` : 'bin\\run'}" %*
+  "%~dp0\\..\\client\\bin\\fluence.cmd" %*
 )
 `,
     nsis: ({ arch, config, customization, defenderOptional, hideDefenderOption, }) => `!include MUI2.nsh
@@ -268,7 +269,7 @@ the CLI should already exist in a directory named after the CLI that is the root
         const nsisCustomization = config.nsisCustomization ? (0, node_fs_1.readFileSync)(config.nsisCustomization, 'utf8') : '';
         const arches = buildConfig.targets.filter((t) => t.platform === 'win32').map((t) => t.arch);
         await Tarballs.build(buildConfig, {
-            pack: false,
+            pack: true,
             parallel: true,
             platform: 'win32',
             pruneLockfiles: flags['prune-lockfiles'],
diff --git a/lib/tarballs/bin.js b/lib/tarballs/bin.js
index 9272ce0ff10b1f09c3c8dbbb7e2ccdf72dc8268e..3ba9b823baf13e7c65ccf76b5f531339bfee7316 100644
--- a/lib/tarballs/bin.js
+++ b/lib/tarballs/bin.js
@@ -57,6 +57,8 @@ if exist "%~dp0..\\bin\\node.exe" (
     const writeUnix = async () => {
         const bin = path.join(baseWorkspace, 'bin', config.bin);
         await fs.promises.writeFile(bin, `#!/usr/bin/env bash
+export NODE_NO_WARNINGS=1
+
 set -e
 echoerr() { echo "$@" 1>&2; }
 
